src/ast/ast_validations.c

```diff
@@ -456,6 +456,19 @@ static AST_Validation _ValidateInlinedPropertiesOnPath(const cypher_astnode_t *p
 	return AST_VALID;
 }
 
+// TODO: Remove this once path patterns are supported at filtering.
+// This function is NOT syntax validation function. It is checks there are path pattern in the filter.
+static AST_Validation _Validate_filter_not_contains_path(const cypher_astnode_t *ast_node) {
+	cypher_astnode_type_t type = cypher_astnode_type(ast_node);
+	if(type == CYPHER_AST_PATTERN_PATH) return AST_INVALID;
+	uint childCount = cypher_astnode_nchildren(ast_node);
+	for(uint i = 0; i < childCount; i++) {
+		if(_Validate_filter_not_contains_path(cypher_astnode_get_child(ast_node, i)) == AST_INVALID)
+			return AST_INVALID;
+	}
+	return AST_VALID;
+}
```

Comment:
This pattern of searching/scanning the AST repeat itself, consider replacing this logic with a more generic function which given a root `cypher_astnode_t root` node and a `cypher_astnode_type_t t` scans the tree beneath node and returns true/false if `t` was found, (we might already have such a function)